[Bug]: Arm builds are broken since update to rust 1.81

### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

#5032 updated rust to 1.81, but [lemmy-cross-toolchains](https://github.com/raskyld/lemmy-cross-toolchains) still [ships rust 1.80](https://github.com/raskyld/lemmy-cross-toolchains/blob/main/aarch64-lemmy-linux-gnu/Dockerfile#L14).

building lemmy with rust 1.80 is not supported, as the codebase uses features only stabilized in 1.81.

### Steps to Reproduce

1. build lemmy for arm

### Technical Details

```
error[E0658]: the `#[expect]` attribute is an experimental feature
   --> crates/db_schema/src/newtypes.rs:194:1
    |
194 | #[expect(clippy::from_over_into)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #54503 <https://github.com/rust-lang/rust/issues/54503> for more information
```

### Version

0.19.6-beta.7

### Lemmy Instance URL

_No response_

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -5,7 +5,7 @@ ARG RUST_RELEASE_MODE=debug
 
 ARG AMD_BUILDER_IMAGE=rust:${RUST_VERSION}
 # Repo: https://github.com/raskyld/lemmy-cross-toolchains
-ARG ARM_BUILDER_IMAGE="ghcr.io/raskyld/aarch64-lemmy-linux-gnu:v0.4.0"
+ARG ARM_BUILDER_IMAGE="ghcr.io/raskyld/aarch64-lemmy-linux-gnu:v0.5.0"
 
 ARG AMD_RUNNER_IMAGE=debian:bookworm-slim
 ARG ARM_RUNNER_IMAGE=debian:bookworm-slim

